src/settings/settingsExport.ts

```diff
@@ -0,0 +1,114 @@
+import GioUnix from 'gi://Gio';
+import GLib from 'gi://GLib';
+import Settings from '@settings/settings';
+
+const dconfPath = '/org/gnome/shell/extensions/tilingshell/';
+const excludedKeys = [
+    Settings.SETTING_LAYOUTS_JSON,
+    Settings.SETTING_LAST_VERSION_NAME_INSTALLED,
+    Settings.SETTING_OVERRIDDEN_SETTINGS,
+];
+
+export default class SettingsExport {
+    dconfDump: string;
+
+    constructor() {
+        this.dconfDump = this.dumpDconf();
+        this.excludeKeys();
+    }
+
+    static importFrom(content: string) {
+        this.restoreToDefault();
+
+        // Flush overridden settings back into system
+        Settings.set_active_screen_edges(false);
+        Settings.set_enable_move_keybindings(false);
```

Comment:
> * Due to the workaround used for overridden settings, if the export file is missing the key `active-screen-edges` or `enable-move-keybindings`, these two options will remain disabled after import, instead of the default enabled state

The reason behind this is that, during the import of settings, the settings are reset and then `active-screen-edges` and `enable-move-keybindings` are set to false.

I see you want to flush all the overridden settings back into system, you could use `SettingsOverride.get().restoreAll()`

Reply:
Appreciate the suggestion!

`SettingsOverride.get().restoreAll()` works except for one situation: If either of these toggles are enabled both before and after the import, the overridden settings remain restored.

I can think of two solutions:
- If the state of either is `enabled` both before and after import:
  a. Apply overrides again (is there a function for this?), or
  b. Toggle it off and on

Reply:
Sorry, what do you mean with "_the overridden settings remain restored_"?

Reply:
The value of `overriden-settings` is only populated when you enable the toggle.

In this case, `SettingsOverride.get().restoreAll()` restores the overridden settings, but it doesn't get populated again since the state of the toggle is unchanged.

Hope that clears things out!

Reply:
Uuuuhhhh! I understand what do you mean...
Thanks a million, I believe that toggling it off and on, as you suggested, could be the best workaround (there isn't a function to override again). Can you put a comment near this workaround to remember these details in the future? Thanks again!